txpool/txpool.go

```diff
@@ -88,15 +96,70 @@ func (p *Pool) Transaction(id sunyata.TransactionID) (sunyata.Transaction, bool)
 	return txn, ok
 }
 
-// Transactions returns the transactions currently in the pool.
+// Transactions returns the transactions currently in the pool. The transaction
+// set is ordered in a way that can acceptably be included in a block; such
+// that all parent transactions occur before their children.
 func (p *Pool) Transactions() []sunyata.Transaction {
```

Comment:
I think this needs to return a `[][]Transaction`. The miner is going to do further processing of its own (e.g. to stay under the block size limit), so without knowing what the parent->child chains are, it runs the risk of accidentally splitting them up.

Reply:
Is it alright to return duplicates? A child with 2 parents would need to be included twice. 
```
A, B -> C
[A, C]
[B, C]
```

Could we update the any remaining unconfirmed transaction's ephemeral outputs with the newly created output in txpool's `ProcessChainApplyUpdate`? That would let the miner add/remove any transactions it wants provided it maintains the initial ordering.

Reply:
Should we also be updating any remaining unconfirmed transaction's ephemeral outputs with the newly created output in txpool's `ProcessChainApplyUpdate`? That way any children that don't make it into a block will still be valid.
```
for i := range txn.Inputs {
	cau.UpdateOutputProof(&txn.Inputs[i].Parent)
	// Update the parent of any ephemeral inputs that are newly confirmed.
	if o, exists := createdOutputs[txn.Inputs[i].Parent.ID]; exists {
		txn.Inputs[i].Parent = o
	}
}
```

Reply:
yep, we should definitely turn ephemeral outputs into standard outputs if their parent appears in a block

Reply:
Updating the parent first causes `UpdateOutputProof` to panic
```
--- FAIL: TestMineBlock (0.01s)
panic: runtime error: slice bounds out of range [5:1] [recovered]
        panic: runtime error: slice bounds out of range [5:1]

goroutine 18 [running]:
testing.tRunner.func1.2(0x1170ee0, 0xc000240000)
        testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000192300)
        testing/testing.go:1146 +0x4b6
panic(0x1170ee0, 0xc000240000)
        runtime/panic.go:965 +0x1b9
go.sia.tech/sunyata/consensus.(*StateApplyUpdate).UpdateOutputProof(0xc000238000, 0xc0001f0ee0)
        go.sia.tech/sunyata/consensus/update.go:67 +0x4af
go.sia.tech/sunyata/txpool.(*Pool).ProcessChainApplyUpdate(0xc0001de000, 0xc000238000, 0x128b700, 0x0, 0x0)
        go.sia.tech/sunyata/txpool/txpool.go:200 +0x4be
go.sia.tech/sunyata/chain.(*Manager).AddTipBlock(0xc0001d7300, 0x1, 0x2c1ef5206b0a2dcf, 0x220e2a39f9ee33d6, 0x6a75490e1cad63a9, 0x581cd8d49f6a1558, 0x105022ac3a76bccc, 0x0, 0xed88d4946, 0x128b7c0, ...)
        go.sia.tech/sunyata/chain/manager.go:365 +0x702
go.sia.tech/sunyata/miner_test.TestMineBlock(0xc000192300)
        go.sia.tech/sunyata/miner_test/miner_test.go:130 +0x12a5
testing.tRunner(0xc000192300, 0x118d178)
        testing/testing.go:1193 +0xef
created by testing.(*T).Run
        testing/testing.go:1238 +0x2b3
FAIL    go.sia.tech/sunyata/miner       0.250s
```

Reply:
sry, what I meant was that it shouldn't be called on ephemeral outputs at all -- they don't have proofs, so there's nothing to update.

We *could* add a simple check in `UpdateOutputProof` to make it no-op on ephemeral outputs; maybe that would be best.